src/corehost/common/pal.windows.cpp

```diff
@@ -253,17 +251,26 @@ bool pal::get_own_executable_path(string_t* recv)
     return true;
 }
 
-pal::string_t pal::to_palstring(const std::string& str)
+pal::string_t pal::utf8_palstring(const std::string& str)
 {
-    return t_converter.from_bytes(str);
+    // No need of explicit null termination, so pass in the actual length.
+    pal::string_t retval;
+    size_t size = ::MultiByteToWideChar(CP_UTF8, 0, str.c_str(), str.size(), nullptr, 0);
+    retval.resize(size, '\0');
```

Comment:
Given that MBTOWC returns zero for failure, I dont think the resize should be done based upon the size. If your goal is to initialize the variable, then please do so explicitly, independent of the call.

Also, if the call failed, we should propagate that failure up instead of returning an empty string!